ninchatsdk/src/main/java/com/ninchat/sdk/ninchatvideointegrations/jitsi/NinchatGroupCallIntegration.kt

```diff
@@ -1,36 +1,213 @@
 import com.ninchat.sdk.ninchatvideointegrations.jitsi.presenter.NinchatGroupCallPresenter
 import com.ninchat.sdk.ninchatvideointegrations.jitsi.presenter.OnClickListener
-import kotlinx.android.synthetic.main.ninchat_join_end_conference.view.*
+import com.ninchat.sdk.utils.keyboard.hideKeyBoardForce
+import kotlinx.android.synthetic.main.activity_ninchat_chat.*
+import kotlinx.android.synthetic.main.activity_ninchat_chat.view.*
+import kotlinx.android.synthetic.main.ninchat_join_end_conference.*
+import kotlinx.android.synthetic.main.ninchat_titlebar.view.*
+import org.jitsi.meet.sdk.BroadcastIntentHelper
+import org.jitsi.meet.sdk.JitsiMeetConferenceOptions
+import org.jitsi.meet.sdk.JitsiMeetUserInfo
+import org.jitsi.meet.sdk.JitsiMeetView
+import java.net.URL
 
 class NinchatGroupCallIntegration(
-    private val videoContainer: View,
+    private val jitsiMeetView: JitsiMeetView,
+    private val mActivity: NinchatChatActivity,
     chatClosed: Boolean = false,
 ) {
-    private val model = NinchatGroupCallModel(endConference = chatClosed).apply {
+    private val model = NinchatGroupCallModel(
+        chatClosed = chatClosed,
+    ).apply {
         parse()
     }
     private val presenter = NinchatGroupCallPresenter(model = model)
     private val onClickListener = OnClickListener(intervalInMs = 2000)
 
     init {
-        presenter.renderView(videoContainer = videoContainer)
+        presenter.renderInitialView(mActivity = mActivity)
         attachHandler()
     }
 
     private fun attachHandler() {
-        videoContainer.conference_join_button.setOnClickListener {
+        mActivity.conference_join_button.setOnClickListener {
             onClickListener.onClickListener {
                 presenter.onClickHandler()
             }
         }
+        val params = FrameLayout.LayoutParams(
+            FrameLayout.LayoutParams.MATCH_PARENT,
+            FrameLayout.LayoutParams.MATCH_PARENT
+        )
+        mActivity.jitsi_frame_layout.addView(jitsiMeetView, params)
     }
 
-    fun updateView(chatClosed: Boolean) {
-        model.update(endConference = chatClosed)
-        presenter.renderView(videoContainer = videoContainer)
+    fun onChannelClosed() {
+        model.chatClosed = true
+        hangUp()
+    }
+
+    fun hangUp() {
+        LocalBroadcastManager.getInstance(mActivity.applicationContext)
+            .sendBroadcast(BroadcastIntentHelper.buildHangUpIntent())
+
+        // wait for the jitsi event to update the view
+        // if jitsi is not running what should we do ? -> if was not running then call onHangup manually to propagate UI updates
+        if (model.onGoingVideoCall) {
+            return
+        }
+        onHangup()
+    }
+
+    fun startJitsi(
+        jitsiRoom: String,
+        jitsiToken: String,
+        jitsiServerAddress: String,
+    ) {
+        val options: JitsiMeetConferenceOptions = JitsiMeetConferenceOptions.Builder()
+            .setRoom("pallab-test-1")
```

Comment:
Should this be jitsiRoom variable?

Reply:
Ah. This is probably on purpose, since we are still using public server and not ours.